### PR TITLE
Gh 446 quick start

### PR DESCRIPTION
**Pull request by maleck13**
**what**

- Updates to add the add-on manager during quickstart and install the kuadrant service protection components into the setup clusters ready to use with deployed gateways

- Allows for running the quick start from a branch
- increases the stale timeout 
- removes some unneeded kustomize directories

**Verficiation**
``` export MGC_BRANCH=GH-446-quick-start```

execute the quick start from this branch

``` 
curl https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/GH-446-quick-start/hack/quickstart-setup.sh | bash
```

Should see output
```setting MGC_REPO to use branch GH-446-quick-start```

Once the setup has completed verify that the add-on manager is present and kuadrant is installed

against the hub execute

```
k get pods -n multicluster-gateway-controller-system
```
you should see ```mgc-kuadrant-add-on-manager``` running

against the hub and spoke execute ```k get pods -n kuadrant-system```

should see the kuadrant service protection components running 

```
authorino-7df5ff9bf7-sr79m                               1/1     Running   0          4m38s
authorino-operator-76555b94b8-r8zwt                      1/1     Running   0          5m2s
kuadrant-operator-controller-manager-6cf7967c8c-nl5sn    1/1     Running   0          5m5s
limitador-54fd6d57-w855z                                 1/1     Running   0          4m34s
limitador-operator-controller-manager-5f4d7bc769-wsxlq   1/1     Running   0          5m
```

### CODE REVIEW

**Comment by jasonmadigan** [comment]
@maleck13 how're you testing this right now? Tried the quickstart setup w/ your branch, but not seeing the Kuadrant bits installed.

```
curl https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/GH-446-quick-start/hack/quickstart-setup.sh | bash
```

**Comment by Ygnas** [comment]
> @maleck13 how're you testing this right now? Tried the quickstart setup w/ your branch, but not seeing the Kuadrant bits installed.
> 
> ```
> curl https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/GH-446-quick-start/hack/quickstart-setup.sh | bash
> ```

export MGC_BRANCH=GH-446-quick-start

At the start of the script, you should see `setting MGC_REPO to use branch GH-446-quick-start`

**Comment by jasonmadigan** [comment]
> 



> > @maleck13 how're you testing this right now? Tried the quickstart setup w/ your branch, but not seeing the Kuadrant bits installed.
> > ```
> > curl https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/GH-446-quick-start/hack/quickstart-setup.sh | bash
> > ```
> 
> export MGC_BRANCH=[GH-446](https://github.com/Kuadrant/multicluster-gateway-controller/issues/446)-quick-start
> 
> At the start of the script, you should see `setting MGC_REPO to use branch GH-446-quick-start`

Good spot - thanks @Ygnas!

**Comment by jasonmadigan** [comment]
@Ygnas were you able to get the QS to work?
Seeing this fairly consistently (with the MGC_BRANCH env var)
```
Error from server (NotFound): error when creating "STDIN": namespaces "kind-mgc-workload-1" not found
Waiting for github.com/kuadrant/multicluster-gateway-controller.git/config/quick-start/control-cluster?ref=GH-446-quick-start cluster config apply... Trying again in 5s
```

**Comment by maleck13** [comment]
> @Ygnas were you able to get the QS to work? Seeing this fairly consistently (with the MGC_BRANCH env var)
> 
> ```
> Error from server (NotFound): error when creating "STDIN": namespaces "kind-mgc-workload-1" not found
> Waiting for github.com/kuadrant/multicluster-gateway-controller.git/config/quick-start/control-cluster?ref=GH-446-quick-start cluster config apply... Trying again in 5s
> ```

I broke this, will ping once fixed

File: hack/quickstart-setup.sh [hunk 1]
```diff
@@ -16,23 +16,42 @@
...
+dockerBinCmd() {
+  echo "docker run --rm -u $UID -v ${TMP_DIR}:/tmp/mgc:z --network mgc -e KUBECONFIG=/tmp/mgc/kubeconfig --entrypoint=$1 $TOOLS_IMAGE"
+}
+
+export KIND_BIN=kind
+export HELM_BIN=helm
 export OPERATOR_SDK_BIN=$(dockerBinCmd "operator-sdk")
 export YQ_BIN=$(dockerBinCmd "yq")
 export CLUSTERADM_BIN=$(dockerBinCmd "clusteradm")
+export KUSTOMIZE_BIN=$(dockerBinCmd "kustomize")
```
Comment (Ygnas) [inline hack/quickstart-setup.sh:31]: Most of these are already imported from `hack/.quickstartEnv`

File: .github/workflows/stale-closer.yaml [hunk 2]
```diff
@@ -13,13 +13,13 @@ jobs:
...
           days-before-issue-close: 30
           stale-issue-label: "stale"
-          stale-issue-message: "This issue is stale because it has been open for 30 days with no activity."
-          stale-pr-message: "This PR is stale because it has been open for 30 days with no activity."
+          stale-issue-message: "This issue is stale because it has been open for 60 days with no activity."
+          stale-pr-message: "This PR is stale because it has been open for 60 days with no activity."
           close-issue-message: "This issue was closed because it has been inactive for 30 days since being marked as stale."
           close-pr-message: "This PR was closed because it has been inactive for 30 days since being marked as stale."
-          days-before-pr-stale: 30
+          days-before-pr-stale: 60
```
Comment (pmccarthy) [inline .github/workflows/stale-closer.yaml:23]: Will this remove the stale label from PRs and issues between 30 days and 60 days? Or should we review and manually update?

Comment (maleck13) [reply .github/workflows/stale-closer.yaml:23]: I don't think it will do it auto but not sure

File: hack/quickstart-setup.sh [hunk 3]
```diff
@@ -16,23 +16,39 @@
...
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/main/hack/.cleanupUtils)"
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/main/hack/.deployUtils)"
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/main/hack/.startUtils)"
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/main/hack/.setupEnv)"
+export TOOLS_IMAGE=quay.io/kuadrant/mgc-tools:latest
+export TMP_DIR=/tmp/mgc
+
+dockerBinCmd() {
+  echo "docker run --rm -u $UID -v ${TMP_DIR}:/tmp/mgc:z --network mgc -e KUBECONFIG=/tmp/mgc/kubeconfig --entrypoint=$1 $TOOLS_IMAGE"
+}
```
Comment (Ygnas) [inline hack/quickstart-setup.sh:24]: These could also be removed, but the exports would need to be moved below the sourcing of files.

**Comment by pmccarthy** [comment]
Successfully ran through verification steps. All working as expected. I can see a couple of comments from @Ygnas so I'll hold off on the lgtm for now

**Comment by jasonmadigan** [comment]
Verified the quickstart script works well.

`make local-setup` doesn't seem to install the addon - not quite sure why yet...

File: hack/quickstart-setup.sh (hunk 3 above)
Comment (maleck13) [reply hack/quickstart-setup.sh]: updated now

**Review by Ygnas — APPROVED** [review]
Successfully verified the steps.

**Comment by jasonmadigan** [comment]
Pushed a commit to add the manager for local-setup, i.e. `make local-setup OCM_SINGLE=true MGC_WORKLOAD_CLUSTERS_COUNT=1`.

/lgtm

**Pull request merged**